sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateStoreProvider.scala

```diff
@@ -287,7 +314,8 @@ private[sql] class RocksDBStateStoreProvider
       useColumnFamilies)
   }
 
-  @volatile private var encoder: RocksDBStateEncoder = _
+  @volatile private var keyValueEncoderMap = new java.util.concurrent.ConcurrentHashMap[String,
```

Comment:
nit: could be `val`, and also probably no need to be `@volatile`.

Reply:
Done